dist/api_data.py

```diff
@@ -1138,6 +1138,12 @@ def __ge__(self, other):
     wiredtiger_open_log_configuration +\
     wiredtiger_open_tiered_storage_configuration +\
     wiredtiger_open_statistics_log_configuration + [
+    Config('backup_restore_target', '', r'''
+        If non-empty and restoring from a backup, restore only the table object targets listed.
```

Comment:
From my understanding @kommiharibabu, I think we do require that we only do a partial restore because the current implementation does require the metadata backup file. I can see possible ways where we can extend this functionality to look at the current state of the metadata file instead, but I think that benefit should be completed in a follow-up ticket. What are your thoughts?

Reply:
I realised that this might be actually a cool idea we can extend @sueloverso  @agorrod what are your thoughts on this?

Reply:
I agree - it's an interesting idea, but would require more work, since the metadata manipulation would need to happen in a different place. It does make me wonder if we should update the name of the configuration option, so that we can make that change in the future.
I'd leave that one to @sueloverso, since her naming game is strong, and she understands much better how easily this would be extended to work without a backup, which probably influences how hard we should try to make the configuration option name generic.

Reply:
I don't think there is a future for people manually removing files from an active database and wanting to restore and run it partially. We strongly discourage them from manipulating the filespace and backup is really the only avenue to legitimately do that. 

I could see expanding the backup `target=` API or meaning to include WT files so that you get a working smaller database from the backup cursor. MongoDB cannot do that because at the "take the backup" time they don't know what they need.

If the partial restore is expanded then it will be a real project and its API can be considered at that time. Modifying the API at that point would be fine.

Reply:
Makes sense! It sounds like there won't be much benefit implementing this outside of backup. I am resolving this conversation